redis/deployment/src/test/java/io/quarkiverse/langchain4j/redis/deployment/RedisEmbeddingStoreTest.java

```diff
@@ -0,0 +1,232 @@
+package io.quarkiverse.langchain4j.redis.deployment;
+
+import static dev.langchain4j.internal.Utils.randomUUID;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.Percentage.withPercentage;
+
+import java.util.List;
+
+import jakarta.inject.Inject;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.AllMiniLmL6V2QuantizedEmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.CosineSimilarity;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.RelevanceScore;
+import io.quarkiverse.langchain4j.redis.RedisEmbeddingStore;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class RedisEmbeddingStoreTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest unitTest = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource(new StringAsset("quarkus.langchain4j.redis.dimension=384\n" +
+                            "quarkus.redis.devservices.image-name=redis/redis-stack:latest\n" +
```

Comment:
So I assume the default image we use with the Redis extension does not provide the necessary features?

Reply:
Yeah, the default image doesn't have RediSearch. Perhaps we could try to detect if we're using an unsuitable image and tell the user....

Reply:
Right, my concern is that if we don't help people, they will have no idea they need to use this in order to get devservices for this feature working

Reply:
No. You need "search" which is only available in redis-stack. Redis-stack is not the default as it contains non-opensource modules, such as "search", "json" ...

Reply:
Yeah, but how would users know they have to configure it?

Reply:
I am wondering whether the extension should default to this image? We can likely do it by providing a configuration value with lower priority than the user configured property

Reply:
I've added some detection logic that throws an error and suggests to use `redis/redis-stack` images

Reply:
It's still a bit uncomfortable that when you're using dev services, you have to add `quarkus.redis.devservices.image-name=redis/redis-stack` .. Would it be possible and reasonable to automatically change the default Redis devservices image in case that we have an Embedding store using it?

Reply:
> I've added some detection logic that throws an error and suggests to use redis/redis-stack images

👍🏼

Reply:
@cescoffier we could add a `META-INF/microprofile-config.properties` straight into the extension that specifies a value for `quarkus.redis.devservices.image-name`. But what version should it point at? The redis extension defaults to `redis:7`, but redis-stack doesn't have a `7` tag. Using `latest` sounds a bit dangerous. WDYT?